multiversx_sdk/network_providers/api_network_provider.py

```diff
@@ -300,9 +315,17 @@ def _do_get(self, url: str) -> Any:
         except Exception as err:
             raise NetworkProviderError(url, err)
 
+    def _remove_unneeded_options_for_request(self) -> dict[str, Any]:
```

Comment:
This hack / workaround can be removed if we use the approach with separate retry options.

Reply:
removed